### PR TITLE
[doc] Fix documentation of options onlynet and listenonion

### PR DESCRIPTION
**Pull request by ghost**
See #13378
I think a first step to improvement is to document precisely and correctly the current behavior, and then decide if that behavior is wanted and/or accepted.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -450,14 +450,14 @@ void SetupServerArgs(NodeContext& node)
...
     argsman.AddArg("-maxuploadtarget=<n>", strprintf("Tries to keep outbound traffic under the given target (in MiB per 24h). Limit does not apply to peers with 'download' permission. 0 = no limit (default: %d)", DEFAULT_MAX_UPLOAD_TARGET), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-onion=<ip:port>", "Use separate SOCKS5 proxy to reach peers via Tor onion services, set -noonion to disable (default: -proxy)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-onlynet=<net>", "Make outgoing connections only through network <net> (ipv4, ipv6 or onion). Incoming connections are not affected by this option. This option can be specified multiple times to allow multiple networks.", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-onlynet=<net>", "Make outgoing connections only through network <net> (ipv4, ipv6 or onion). Incoming connections are not affected by this option. This option can be specified multiple times to allow multiple networks. Regardless of this setting, outgoing onion connections are enabled by listenonion=1", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
Comment (laanwj) [inline src/init.cpp:460]: Nit: missing `.` at end of sentence

**Comment by wodry** [comment]
Added the dot, rebased&squashed.

**Comment by wodry** [comment]
I would appreciate if this could be merged now, so it is done.

**Comment by wodry** [comment]
The chaotic networking behavior is now reasonably documented with #20757 I guess, closing.

**Comment by jonatack** [comment]
@wodry I'm sorry, I overlooked this pull when consolidating ideas for #20757, otherwise would have been happy to look at including it. Maybe it's complementary.

**Comment by wodry** [comment]
@jonatack Thanks, that's nice of you, nevermind, my impression is that the subject of Tor network and it's documentation is in good hands of you right now. There was some dynamic changes currently in this subject, I lost overview and only use release versions, so I do not know the current (master branch) behavior. I would check later time with new releases maybe again. Feel free to check if the issue I mentioned here is of any value anymore maybe worth to include :-)

**Pull request closed without merging**